extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -149,16 +150,58 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
       return;
     }
 
-    this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
     this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
-    this.mutateElement(() => {
-      createShadowRootWithStyle(
-        this.element,
-        this.shoppingTagEl_,
-        shoppingTagCSS
-      );
-      this.hasAppendedInnerShoppingTagEl_ = true;
-    });
+
+    this.measureMutateElement(
+      () => {
+        this.shoppingTagEl_ = this.renderShoppingTagTemplate_();
+      },
+      () => {
+        createShadowRootWithStyle(
+          this.element,
+          this.shoppingTagEl_,
+          shoppingTagCSS
+        );
+        this.hasAppendedInnerShoppingTagEl_ = true;
+
+        const rightEdgeOfScreentoTagDistance =
+          document.getElementsByTagName('amp-story-shopping-attachment')[0]
+            ./*OK*/ clientWidth *
+            0.5 -
+          this.element./*OK*/ offsetLeft -
+          this.element./*OK*/ offsetWidth * 0.5;
+
+        const shoppingTagInner = this.element.shadowRoot.querySelector(
+          '.amp-story-shopping-tag-inner'
+        );
+
+        if (this.shoppingTagEl_.getAttribute('dir') !== 'rtl') {
```

Comment:
To recap our explorations:
It looks like RTL state is set when the `body` or `html` element has the `dir` attribute.
A separate RTL example document will be necessary to test RTL that has `dir="rtl"` on `body` or `html`.